src/oneapi/oneapi.pyx

```diff
@@ -45,25 +45,30 @@ cdef class sycl_execution_context:
         del self.c_ptr
 
 
-from contextlib import contextmanager
+# thread-local storage
+from threading import local as threading_local
+_tls = threading_local()
+if getattr(_tls, 'initialized', None) is None:
+    _tls._in_sycl_ctxt = False
+    _tls.initialized = True
 
+def is_in_sycl_ctxt():
+    return _tls._in_sycl_ctxt
 
-# Not thread-safe or re-entrant or anything...
-cdef bool _in_sycl_ctxt = False
 
+from contextlib import contextmanager
 
 @contextmanager
 def sycl_context(dev='default'):
-    global _in_sycl_ctxt
     # Code to acquire resource
     ctxt = sycl_execution_context(dev)
-    _in_sycl_ctxt = True
+    _tls._in_sycl_ctxt = True
```

Comment:
Same thing here. `_tls` may not have `initialized` variable set. 

I think you should write and use a setter routine which checks whether `_tls` has been initialized and initializes it if needed.

Reply:
done